strolle-gpu/src/utils.rs

```diff
@@ -41,3 +41,26 @@ pub fn resolve_checkerboard_alt(global_id: UVec2, frame: u32) -> UVec2 {
 pub fn got_checkerboard_at(screen_pos: UVec2, frame: u32) -> bool {
     screen_pos == resolve_checkerboard(screen_pos / uvec2(2, 1), frame)
 }
+
+pub fn safe_normalize(v: Vec3) -> Vec3 {
```

Comment:
Good spot, I have reverted all safe_normalize back to their original form. I should try to explain the issue I had, I now think the compile issue was due to specifically any_orthonormal_pair, for example:

```
pub fn safe_any_orthonormal_pair(normal: Vec3) -> (Vec3, Vec3) {
    normal.any_orthonormal_pair()
}
```

Leads to:

```
Shader validation error:
  ┌─ C:\...\di_sampling-main.spv:1:1
  │
1 │
  │   naga::Expression [73]


    Constant expression [73] is invalid
    Float literal is NaN

```

Now I don't know why wgpu decided to specifically complain about this. It seems a little arbitrary as I'm sure there are Inf and NaN generated elsewhere that do not cause problems. I even tried to disassemble the spir-v to see if there was a NaN constant explicitly generated somewhere, but I couldn't see anything obvious. Also it could be a combination of issues leading to the compiler producing a NaN constant. Either way, I solved the issue by being overly cautious with zero constants and divisions.